src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -1703,17 +1703,14 @@ private static bool IsPropertyAssignedThroughBackingField(BoundExpression receiv
                 propertySymbol = propertySymbol.OriginalDefinition;
             }
 
-            // PROTOTYPE(semi-auto-props): TODO: Support assigning semi auto prop from constructors.
-
             return propertySymbol is SourcePropertySymbolBase sourceProperty &&
-                    // PROTOTYPE(semi-auto-props): Consider `public int P { get => field; set; }`
-                    sourceProperty.GetMethod is SourcePropertyAccessorSymbol { IsEquivalentToBackingFieldAccess: true } &&
+                    IsConstructorOrField(fromMember, isStatic: sourceProperty.IsStatic) &&
+                    (sourceProperty.GetMethod is SourcePropertyAccessorSymbol { IsEquivalentToBackingFieldAccess: true } || sourceProperty.FieldKeywordBackingField is not null) &&
                     // To be assigned through backing field, either SetMethod is null, or it's equivalent to backing field write
                     // PROTOTYPE(semi-auto-props): TODO: Do we need to use `GetOwnOrInheritedSetMethod` instead of `SetMethod`?
                     // Legacy auto-properties are required to override all accessors. If this is not the case with semi auto props, we may need to use GetOwnOrInheritedSetMethod
                     sourceProperty.SetMethod is null or SourcePropertyAccessorSymbol { IsEquivalentToBackingFieldAccess: true } &&
```

Comment:
>sourceProperty.SetMethod is null or SourcePropertyAccessorSymbol { IsEquivalentToBackingFieldAccess: true } [](http://example.com/codeflow?start=20&length=107)

Consider checking the setter first since this check doesn't depend on presence of the `field` keyword.

Reply:
It looks like this reduced some extra bindings! Thanks @AlekseyTs!